packages/lib/ren-tx/src/configs/genericMint.ts

```diff
@@ -62,25 +63,35 @@ const txCreator = async (context: GatewayMachineContext) => {
         context.tx.nonce = RenJS.utils.randomNonce().toString("hex");
     }
 
+    const { targetAmount, sourceAsset, sourceChain, destChain } = context.tx;
+
+    const to = context.toChainMap[destChain](context);
+    const from = context.fromChainMap[sourceChain](context);
+
+    const decimals = await from.assetDecimals(sourceAsset.toUpperCase());
+
+    let suggestedAmount = new BigNumber(Number(targetAmount) * 10 ** decimals);
```

Comment:
The maths should probably happen at the BigNumber level to avoid losing precision.

```suggestion
    let suggestedAmount = new BigNumber(Number(targetAmount)).times(new BigNumber(10).exponentiatedBy(decimals));
```